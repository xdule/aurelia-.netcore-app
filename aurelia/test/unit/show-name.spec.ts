import { bootstrap } from 'aurelia-bootstrapper';
import { StageComponent } from 'aurelia-testing';
import { PLATFORM } from 'aurelia-framework';

describe('ShowNameComponent', () => {
  let component;

  beforeEach(async () => {
    component = StageComponent
      .withResources([
        PLATFORM.moduleName('../../src/app/resources/elements/show-name/show-name.element'),
      ])
      .inView('<show-name some-text.bind="firstName"></show-name>')
      .boundTo({ firstName: 'Bob' });
      await component.create(bootstrap);
  });

  it('should render correctly', () => {
    expect(document.body.outerHTML).toMatchSnapshot();
  });

  it('should render first name', () => {
    const nameElement = document.querySelector('#fn') as HTMLInputElement;
    expect(nameElement.innerHTML).toBe('Bob');
  });

  afterEach(() => {
    component.dispose();
  });
});
