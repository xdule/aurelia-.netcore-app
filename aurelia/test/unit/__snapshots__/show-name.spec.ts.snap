// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowNameComponent should render correctly 1`] = `
"<body><div><show-name some-text.bind=\\"firstName\\" class=\\"au-target show-name\\" au-target-id=\\"2\\">
  <h3>Custom global element</h3>
  <p>Your text: <span id=\\"fn\\">Bob</span></p>
</show-name></div></body>"
`;
